pkg/auth/service.go

```diff
@@ -474,299 +473,275 @@ func (s *DBAuthService) ListEffectivePolicies(ctx context.Context, username stri
+	if _, err := s.GetUser(ctx, username); err != nil {
+		return fmt.Errorf("%s: %w", username, err)
+	}
+	if _, err := s.GetGroup(ctx, groupDisplayName); err != nil {
+		return fmt.Errorf("%s: %w", groupDisplayName, err)
 	}
-	result, err := s.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		if _, err := getUser(tx, username); err != nil {
-			return nil, err
-		}
-		dbGroups := make([]*model.DBGroup, 0)
-		err := tx.Select(&dbGroups, `
-			SELECT auth_groups.* FROM auth_groups
-				INNER JOIN auth_user_groups ON (auth_groups.id = auth_user_groups.group_id)
-				INNER JOIN auth_users ON (auth_user_groups.user_id = auth_users.id)
-			WHERE
-				auth_users.display_name = $1
-				AND auth_groups.display_name > $2
-				AND auth_groups.display_name like $3
-			ORDER BY auth_groups.display_name
-			LIMIT $4`,
-			username, params.After, fmt.Sprint(params.Prefix, "%"), params.Amount+1)
-		if err != nil {
-			return nil, err
-		}
-		groups := make([]*model.Group, len(dbGroups))
-		for i := range dbGroups {
-			groups[i] = model.ConvertGroup(dbGroups[i])
-		}
-		p := &model.Paginator{}
-		if len(groups) == params.Amount+1 {
-			// we have more pages
-			groups = groups[0:params.Amount]
-			p.Amount = params.Amount
-			p.NextPageToken = groups[len(dbGroups)-1].DisplayName
-			return &res{groups, p}, nil
-		}
-		p.Amount = len(groups)
-		return &res{groups, p}, nil
-	})
 
+	gu := model.KvUserToGroup(username, groupDisplayName)
+	err := s.store.DeleteMsg(ctx, model.PartitionKey, gu)
 	if err != nil {
-		return nil, nil, err
+		return fmt.Errorf("remove user from group: (key %s): %w", gu, err)
 	}
-	return result.(*res).groups, result.(*res).paginator, nil
+	return err
 }
 
-func (s *DBAuthService) ListGroupUsers(ctx context.Context, groupDisplayName string, params *model.PaginationParams) ([]*model.User, *model.Paginator, error) {
-	type res struct {
-		users     []*model.User
-		paginator *model.Paginator
+func (s *KVAuthService) ListUserGroups(ctx context.Context, username string, params *model.PaginationParams) ([]*model.Group, *model.Paginator, error) {
+	if _, err := s.GetUser(ctx, username); err != nil {
+		return nil, nil, fmt.Errorf("%s: %w", username, err)
 	}
-	result, err := s.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		if _, err := getGroup(tx, groupDisplayName); err != nil {
-			return nil, err
-		}
-		dbUsers := make([]*model.DBUser, 0)
-		err := tx.Select(&dbUsers, `
-			SELECT auth_users.* FROM auth_users
-				INNER JOIN auth_user_groups ON (auth_users.id = auth_user_groups.user_id)
-				INNER JOIN auth_groups ON (auth_user_groups.group_id = auth_groups.id)
-			WHERE
-				auth_groups.display_name = $1
-				AND auth_users.display_name > $2
-				AND auth_users.display_name like $3
-			ORDER BY auth_groups.display_name
-			LIMIT $4`,
-			groupDisplayName, params.After, fmt.Sprint(params.Prefix, "%"), params.Amount+1)
+
+	hasMoreGroups := true
+	nextPage := ""
```

Comment:
nextPage should be declared above his usage, after the for loop

Reply:
Irrelevant, not the latest version.